contracts/ForwarderV4.sol

```diff
@@ -0,0 +1,353 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+import '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol';
+import '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Receiver.sol';
+import './ERC20Interface.sol';
+import './TransferHelper.sol';
+import './IForwarder.sol';
+
+/**
+ * @title ForwarderV4
+ * @notice This contract will forward any incoming Ether or token to the parent address of the contract
+ */
+contract ForwarderV4 is IERC721Receiver, ERC1155Receiver, IForwarder {
+  /// @notice Any funds sent to this contract will be forwarded to this address
+  address public parentAddress;
+  /// @notice Address which is allowed to call methods on this contract alongwith the parentAddress
+  address public feeAddress;
+  bool public autoFlush721 = true;
+  bool public autoFlush1155 = true;
```

Comment:
What is the difference in cost of forwarder creation with these nft features vs without nft features?

Reply:
Gaslimit used for [Forwarder Creation](https://goerli.arbiscan.io/tx/0xeb896d48fb1b3f1ae000c130d6c80589acfcd226700afb93c2c92c63a3855d01) with NFT auto flushing features: 121,760
Gaslimit used for [Forwarder Creation](https://goerli.arbiscan.io/tx/0x8eda40ff7df6b4387dbb9ebb55f1d6e681ef29275ed1e266ce2d50478ba2d2eb) without NFT auto flushing features: 117,656

Reply:
I cant access the links, but what I would like to see is the difference in gas used (not gaslimit) between forwarder with nft features and without. The features are more than just autoflushing

Reply:
Sorry for the links, at that time i had used goerli arbitrum for the tests and seems like it is not accessible now. I will add the difference without all the nft related methods and with them

Reply:
ok thanks for the comparison. It is weird that arbitrum gas is twice than of optimism. Anyway, not a blocker but something to keep in mind in the future